src/molecule/generate.py

```diff
@@ -0,0 +1,359 @@
+    otherwise False.
+
+    :param path: A file-system path referring to a file or dir that
+        should be rendered or just copied.
+    :param context: cookiecutter context.
+    """
+    try:
+        for dont_render in context["cookiecutter"]["_copy_without_render"]:
+            if fnmatch.fnmatch(path, dont_render):
+                return True
+    except KeyError:
+        return False
+
+    return False
+
+
+def apply_overwrites_to_context(context, overwrite_context):
+    """Modify the given context in place based on the overwrite_context."""
+    for variable, overwrite in overwrite_context.items():
+        if variable not in context:
+            # Do not include variables which are not used in the template
+            continue
+
+        context_value = context[variable]
+
+        if isinstance(context_value, list):
+            # We are dealing with a choice variable
+            if overwrite in context_value:
+                # This overwrite is actually valid for the given context
+                # Let's set it as default (by definition first item in list)
+                # see ``cookiecutter.prompt.prompt_choice_for_config``
+                context_value.remove(overwrite)
+                context_value.insert(0, overwrite)
+            else:
+                raise ValueError(
+                    f"{overwrite} provided for choice variable {variable}"
+                    f"but the choices are {context_value}."
+                )
+        else:
+            # Simply overwrite the value for this variable
+            context[variable] = overwrite
+
+
+def generate_context(
+    context_file="cookiecutter.json", default_context=None, extra_context=None
+):
+    """Generate the context for a Cookiecutter project template.
+
+    Loads the JSON file as a Python object, with key being the JSON filename.
+
+    :param context_file: JSON file containing key/value pairs for populating
+        the cookiecutter's variables.
+    :param default_context: Dictionary containing config to take into account.
+    :param extra_context: Dictionary containing configuration overrides
+    """
+    context = OrderedDict([])
+
+    try:
+        with open(context_file, encoding="utf-8") as file_handle:
+            obj = json.load(file_handle, object_pairs_hook=OrderedDict)
+    except ValueError as e:
+        # JSON decoding error.  Let's throw a new exception that is more
+        # friendly for the developer or user.
+        full_fpath = os.path.abspath(context_file)
+        json_exc_message = str(e)
+        our_exc_message = (
+            f'JSON decoding error while loading "{full_fpath}".  Decoding'
+            f' error details: "{json_exc_message}"'
+        )
+        raise ContextDecodingException(our_exc_message)
+
+    # Add the Python object to the context dictionary
+    file_name = os.path.split(context_file)[1]
+    file_stem = file_name.split(".")[0]
```

Comment:
why not using something like (untested) ? :
```
file_name = os.path.basename(context_file)
file_stem = file_name.splitext(file_name)[0]
```

Reply:
Hi @apatard 

Thanks for your review.

It is referred to [cookiecutter generater.py](https://github.com/cookiecutter/cookiecutter/blob/682fe854dca3ce735d656a6588e9ea9369c4ea47/cookiecutter/generate.py#L105-L107)

They do the same thing getting the filename without extension.

Reply:
Hi @apatard 

`file_name` is generated by `os.path.basename(context_file)` is `string`, which does not have `splitext` function.

So I only change the `file_name` from `os.path.split(context_file)[1]` to `os.path.basename(context_file)`.

Reply:
I think @apatard meant `os.path.splitext(file_name)[0]` and not `file_name.splitext(file_name)[0]`. While I think that would be better, it is a potentially breaking change, since the current code splits at the *first* `.`, while `os.path.splitext` splits at the *last* `.`.

Reply:
Hi @felixfontein 

Thanks for your review. You are right.

The `file_stem ` is generated by `os.path.splitext(file_name)[0]` in new commit.

For this specific case, as `cookiecutter.json` is used, the value of `file_stem` would be `cookiecutter`.